lib/active_merchant/billing/gateways/loan_payment_pro.rb

```diff
@@ -0,0 +1,244 @@
+      end
+
+      def store(credit_card, options = {})
+        post = {}
+        add_payment(post, credit_card)
+        add_address(post, credit_card, options)
+
+        commit(path(:store), post)
+      end
+
+      def unstore(authorization, options = {})
+        commit(path(:unstore, authorization), {}, :delete)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Transactionkey: )\S+), '\1[FILTERED]').
+          gsub(%r((CardNumber=)\d+), '\1[FILTERED]').
+          gsub(%r((CardCode=)\d+), '\1[FILTERED]')
+      end
+
+      private
+
+      def address_from_options(options)
+        options[:billing_address] || options[:address] || {}
+      end
+
+      def address_names(address_name, payment_method)
+        names = split_names(address_name)
+        return names if names.any?(&:present?)
+
+        [payment_method&.first_name, payment_method&.last_name]
+      end
+
+      def add_address(post, payment_method, options)
+        address = address_from_options(options)
+        first_name, last_name = address_names(address[:name], payment_method)
+
+        post[:BillingFirstName] = first_name
+        post[:BillingLastName] = last_name
+        post[:BillingAddress1] = address[:address1]
+        post[:BillingAddress2] = address[:address2]
+        post[:BillingCity] = address[:city]
+        post[:BillingState] = address[:state]
+        post[:BillingZip] = address[:zip]
+      end
+
+      def add_invoice(post, money, options)
+        post[:InvoiceId] = options[:order_id]
+        post[:Amount] = amount(money)
+      end
+
+      def add_payment(post, payment)
+        case payment
+        when CreditCard
+          post[:CardNumber] = payment.number
+          post[:CardCode] = payment.verification_value
+          post[:ExpMonth] = format(payment.month, :two_digits)
+          post[:ExpYear] = format(payment.year, :two_digits)
+        end
+      end
+
+      def parse(body)
+        JSON.parse(body).with_indifferent_access
+      rescue JSON::ParserError, TypeError => e
+        {
+          errors: body,
+          status: 'Unable to parse JSON response',
+          message: e.message
+        }.with_indifferent_access
+      end
+
+      def path(action, value = '')
+        {
+          purchase: 'v2-3/paymentcards/run',
+          authorize: 'v2-3/paymentcards/authorize',
+          capture: 'v2-1/transactions/{placeholder}/capture',
+          void: 'v2/payments/{placeholder}/void',
+          refund: 'v2/payments/{placeholder}/refund',
+          verify: 'v2/paymentcards/validate/detailed',
+          store: 'v2/paymentcards/add',
+          unstore: 'v2/paymentcards/{placeholder}/delete',
+          authorize_with_token: 'v2-3/payments/paymentcards/{placeholder}/authorize',
+          purchase_with_token: 'v2-3/payments/paymentcards/{placeholder}/run'
+        }[action]&.gsub('{placeholder}', value.to_s.split('|').first || '')
```

Comment:
Any particular advantage of using `gsub` over string interpolation with `#{placeholder}`?

Also, the `value` argument implies a string but it looks like the places where this method is called the 2nd argument?

Might be just be me that's getting a little confused by this method, but maybe this would be a good one to have a comment that explains what the expected behavior is here.

Reply:
@naashton you are right, I was trying to avoid Rubocop warnings when using string formatting,  but just using string interpolation is way simpler.